src/layouts/JobPost.astro

```diff
@@ -9,7 +9,8 @@ const jobs = sortedPosts(await Astro.glob('../content/jobs/*.{md,mdx}')).filter(
   return !job.file.split('/').pop().startsWith('_')
 })
 
-const currentPath = Astro.url.pathname
+const pathname = Astro.url.pathname
+const currentPath = pathname.endsWith('/') ? pathname.slice(0, -1) : pathname
```

Comment:
The cause of the issue was because of this `Astro.url.pathname` has a trailing slash(`/`) in the build mode. (It does not in development mode) So updated the logic to take it into account.